### PR TITLE
Write tests for untested code.

### PR DESCRIPTION
**Pull request by khalford**
Cleaned up tests. Renamed files and refactored file structure. Moved some methods to more appropriate locations.

### CODE REVIEW

File: Pynetbox_Data_Uploader/lib/utils/csv_to_dict.py
Comment (meoflynn) [inline Pynetbox_Data_Uploader/lib/utils/csv_to_dict.py:44]: I think this docstring is half complete

File: Pynetbox_Data_Uploader/tests/test_csv_to_dict.py
Comment (meoflynn) [inline Pynetbox_Data_Uploader/tests/test_csv_to_dict.py:25]: `pandas`

File: Pynetbox_Data_Uploader/tests/test_netbox_check.py
Comment (meoflynn) [inline Pynetbox_Data_Uploader/tests/test_netbox_check.py:7]: Could we add a docstring to this method?

File: Pynetbox_Data_Uploader/tests/test_netbox_create.py
Comment (meoflynn) [inline Pynetbox_Data_Uploader/tests/test_netbox_create.py:7]: Could we add a docstring to this method?

File: Pynetbox_Data_Uploader/tests/test_netbox_data.py
Comment (meoflynn) [inline Pynetbox_Data_Uploader/tests/test_netbox_data.py:7]: Could we add a docstring to describe this fixture method?

**Review by meoflynn — CHANGES_REQUESTED** [review]

File: Pynetbox_Data_Uploader/.pylintrc
Comment (DavidFair) [inline Pynetbox_Data_Uploader/.pylintrc:14]: We should drop this extra disable and add `__init__` files to the modules that are missing them

File: Pynetbox_Data_Uploader/lib/netbox_api/netbox_check.py
Comment (DavidFair) [inline Pynetbox_Data_Uploader/lib/netbox_api/netbox_check.py:9]: Can we type `NetBox` e.g.:
```
def __init__(self, netbox: Netbox = None)
```

Additionally you've marked it as optional, but if a user passes `None` we don't handle it below?

File: Pynetbox_Data_Uploader/lib/netbox_api/netbox_create.py
Comment (DavidFair) [inline Pynetbox_Data_Uploader/lib/netbox_api/netbox_create.py:10]: Ditto

File: Pynetbox_Data_Uploader/lib/netbox_api/netbox_data.py
Comment (DavidFair) [inline Pynetbox_Data_Uploader/lib/netbox_api/netbox_data.py:17]: ^

File: Pynetbox_Data_Uploader/lib/utils/csv_to_dict.py
Comment (DavidFair) [inline Pynetbox_Data_Uploader/lib/utils/csv_to_dict.py:18]: Format dict shouldn't do anything which requires the netbox API....

File: Pynetbox_Data_Uploader/lib/utils/csv_to_dict.py
Comment (DavidFair) [inline Pynetbox_Data_Uploader/lib/utils/csv_to_dict.py:54]: Can we do this in something like the `NetBox` class?

File: Pynetbox_Data_Uploader/lib/utils/csv_to_dict.py
Comment (DavidFair) [inline Pynetbox_Data_Uploader/lib/utils/csv_to_dict.py:1]: This file should be called `format_dict` to match the name of the class inside

**Review by DavidFair — CHANGES_REQUESTED** [review]

File: Pynetbox_Data_Uploader/.pylintrc
Comment (khalford) [reply Pynetbox_Data_Uploader/.pylintrc:14]: Yep, didn't realise I needed those. Fixed a while ago, never removed the disable.

File: Pynetbox_Data_Uploader/lib/utils/csv_to_dict.py
Comment (khalford) [reply Pynetbox_Data_Uploader/lib/utils/csv_to_dict.py]: It doesn't do anything with Netbox. It just passes the api object through to the netbox_data.py . This allows you to initialise the api once at the start of a script and pass the api around from there, Rather than moving the url and tokens around and initialising a new api object for every method/class.

**Comment by khalford** [comment]
Pylint is going to fail on this branch due to the .pylintrc edit. I have already fixed these issues in another branch which I will merge once this branch merges.

File: Pynetbox_Data_Uploader/lib/utils/csv_to_dict.py
Comment (DavidFair) [reply Pynetbox_Data_Uploader/lib/utils/csv_to_dict.py]: We need to add a file called `__init__.py` to utils to resolve the import error, since this is a module but we're missing the module "marker file"

Comment (khalford) [reply Pynetbox_Data_Uploader/lib/utils/csv_to_dict.py]: utils does an `__init__.py` already in the directory.

**Comment by DavidFair** [comment]
Sorry I sent you down the wrong track:

```
 tests/test_csv_to_dict.py:2: in <module>
    from utils.csv_to_dict import FormatDict
 ```

The test file name (csv_to_dict.py) and the import needs fixing for this case, to match your rename to `format_dict.py`

Once that's done I'm happy to pair program the pylint error, as suppressing it means it's not actually checking that file

**Review by DavidFair — APPROVED** [review]

**Review by meoflynn — APPROVED** [review]

**Pull request merged**
